src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -329,6 +329,28 @@ def get_node_taints(self) -> Union[List[str], None]:
         # this parameter does not need validation
         return node_taints
 
+    def get_node_initialization_taints(self) -> Union[List[str], None]:
+        """Obtain the value of node_initialization_taints.
+
+        :return: empty list, list of strings or None
+        """
+        # read the original value passed by the command
+        node_init_taints = self.raw_param.get("nodepool_initialization_taints")
+        # normalize, default is an empty list
+        if node_init_taints is not None:
+            node_init_taints = [x.strip() for x in (node_init_taints.split(",") if node_init_taints else [""])]
+        # keep None as None for update mode
+        if node_init_taints is None and self.decorator_mode == DecoratorMode.CREATE:
+            node_init_taints = []
```

Comment:
In create mode, with input `None`, the return value is `[]` (instead of `None`), so you'll need to update the mock data in failed unit test cases (e.g., `ground_truth_mc_1/agent_pool_profiles[0]`).

> FAILED src/aks-preview/azext_aks_preview/tests/latest/test_agentpool_decorator.py::AKSPreviewAgentPoolAddDecoratorStandaloneModeTestCase::test_construct_agentpool_profile_preview
> FAILED src/aks-preview/azext_aks_preview/tests/latest/test_agentpool_decorator.py::AKSPreviewAgentPoolAddDecoratorManagedClusterModeTestCase::test_construct_agentpool_profile_preview
> FAILED src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py::AKSPreviewManagedClusterCreateDecoratorTestCase::test_construct_mc_profile_preview
> FAILED src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py::AKSPreviewManagedClusterCreateDecoratorTestCase::test_set_up_agentpool_profile

https://dev.azure.com/azclitools/public/_build/results?buildId=146099&view=logs&j=59e8686e-5e74-514d-6cad-f7d66c66b425&t=20180a09-30e4-5d69-fc0c-4652edaa2d05&l=2243